generator.py

```diff
@@ -408,66 +167,74 @@ def generate_samples(grammar_dir, outfiles):
     jsgrammar.add_import('cssgrammar', cssgrammar)
 
     for outfile in outfiles:
-        result = generate_new_sample(template, htmlgrammar, cssgrammar,
-                                     jsgrammar)
-
-        if result is not None:
-            print('Writing a sample to ' + outfile)
-            try:
-                f = open(outfile, 'w')
+        result = generate_new_sample(template, htmlgrammar, cssgrammar, jsgrammar)
```

Comment:
Why did you remove the `if result is not None:` check?

Reply:
The reason being that there is almost no possibility of returning None with the current template or if any custom template is being used , as long as it follows the documentation, the generator shouldnt generate a None result, im adding that back in the next commit if you want

Reply:
It appears you are right and it can't return None (maybe it could before at some point).